gcc/rust/resolve/rust-ast-resolve-pattern.h

```diff
@@ -92,8 +134,11 @@ class PatternDeclaration : public ResolverBase
   void visit (AST::RangePattern &pattern) override;
 
 private:
-  PatternDeclaration (Rib::ItemType type) : ResolverBase (), type (type) {}
+  PatternDeclaration (std::vector<PatternBinding> *bindings, Rib::ItemType type)
+    : ResolverBase (), bindings (bindings), type (type)
+  {}
 
+  std::vector<PatternBinding> *bindings = nullptr;
```

Comment:
I guess we can make it work with just references. Does it make a difference?

Reply:
From reading how you use this vector, It doesnt seem like your saving anything by making this a poiinter. Pointers mean the ownership is unclear, because we dont know if this is meant to be free'd or its been allocated at all. We can avoid some of those issues by using const.

For your situation i would turn this into:

```std::vector<PatternBinding> bindings```. And just rely on copy constructors.

Reply:
If you really need it to be persisted and updated across multiple visitors you should use a ```std::vector<PAtternBindings>&``` using a reference is safer because you cant free it but still acts like a pointer so this implies the ownership is up to the initial caller.

Reply:
Ok, I'll go with the reference then.